src/pubsub.ts

```diff
@@ -182,26 +182,29 @@ export class SubscriptionManager {
             // convention this is the value returned from the mutation
             // resolver
             const onMessage = (rootValue) => {
-                let contextPromise;
-                if (typeof options.context === 'function') {
-                    contextPromise = new Promise((resolve) => {
-                        resolve(options.context());
-                    });
-                } else {
-                    contextPromise = Promise.resolve(options.context);
-                }
-                contextPromise.then((context) => {
-                    if (!filter(rootValue, context)) {
+                Promise.resolve().then(() => {
```

Comment:
Woke up with the same thought. Although you can't unnest the `executionResult` otherwise the callback will be called with `(null, undefined)` unless that is desired.